design/trackers.go

```diff
@@ -0,0 +1,125 @@
+package design
+
+import (
+	d "github.com/goadesign/goa/design"
+	a "github.com/goadesign/goa/design/apidsl"
+)
+
+var trackerData = a.Type("TrackerData", func() {
+	a.Attribute("attributes", trackerAttributes)
+	a.Attribute("type", d.String, func() {
+		a.Enum("trackers")
+	})
+	a.Attribute("id", d.String, "ID of the tracker record", func() {
+		a.Example("42")
+	})
+	a.Attribute("links", genericLinks)
+	a.Required("type", "attributes")
+})
+
+var trackerLinks = a.Type("TrackerLinks", func() {
+	a.Attribute("self", d.String, func() {
+		a.Example("http://api.almighty.io/api/trackers/42")
+	})
+	a.Required("self")
+})
+
+// Defines a Media Type for Single Tracker Object
+var TrackerObject = JSONSingle("TrackerObject", "Single Tracker Payload", trackerData, trackerLinks)
+
+var trackerAttributes = a.Type("TrackerAttributes", func() {
+	a.Attribute("url", d.String, "URL of the tracker", func() {
+		a.Example("https://api.github.com/")
+		a.MinLength(1)
+	})
+	a.Attribute("type", d.String, "Type of the tracker", func() {
```

Comment:
As a general question, can we not defer the type from the URL, or URL from the type?